### PR TITLE
[201911 SKU creator] fix wrong speed in case breakout of 2x50 

### PR DESCRIPTION
**Pull request by madhanmellanox**
**- What I did**
I fixed the port split feature of SKU creator tool.

**- How I did it**
When trying to use port split feature of SKU creator tool to split a port in port_config.ini and config_db.json. It was not working, I fixed.

**- How to verify it**
With this fix, we should be able to split and unsplit a port in port_config.ini and config_db.json

**- Previous command output (if the output of a command-line utility has changed)**
No changes to CLI commands

**- New command output (if the output of a command-line utility has changed)**
No changes to CLI commands

### CODE REVIEW

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@madhanmellanox do you need to fix on master as well? if so, master first and then request to cherry pick to 201911.

**Comment by liat-grozovik** [comment]
@abdosi  can you please approve to take to 201911?

**Comment by madhanmellanox** [comment]
retest please

**Comment by madhanmellanox** [comment]
retest please

File: sonic-utilities-tests/sku_create_test.py [hunk 1]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
-            pytest.fail("Output file: {} and model file: {} contents are not same. FAILURE!".format(output_file_path,model_file_path))
+            pytest.fail("Output file: {} and model file: {} contents are not same. FAILURE!".format(output_xml_file_path, model_xml_file_path))
+
+    def test_sku_from_minigraph_file(self):
+        if (os.path.exists(output_minigraph_dir_path)):
+            shutil.rmtree(output_minigraph_dir_path)
+
+        my_command = sku_create_script + " -m "  + minigraph_file  + " -d " + minigraph_input_path
+
+        #Test case execution without stdout
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:100]: Per PEP8, add space after `#`

File: sonic-utilities-tests/sku_create_test.py [hunk 2]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+        result = subprocess.check_output(my_command,stderr=subprocess.STDOUT,shell=True)
+        print result
+
+        #Check if the Output file exists
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:104]: Per PEP8, add space after `#`

File: sonic-utilities-tests/sku_create_test.py [hunk 3]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+        else:
+            pytest.fail("Output file: {} and model file: {} contents are not same. FAILURE!".format(output_minigraph_file_path, model_minigraph_file_path))
+
+    def test_sku_from_config_db_file(self):
+        if (os.path.exists(output_config_db_dir_path)):
+            shutil.rmtree(output_config_db_dir_path)
+
+        my_command = sku_create_script + " -j "  + config_db_file  + " -d " + config_db_input_path
+
+        #Test case execution without stdout
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:122]: Per PEP8, add space after `#`. Do this throughout the file.

File: sonic-utilities-tests/sku_create_test.py [hunk 4]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
 
         #Test case execution without stdout
         result = subprocess.check_output(my_command,stderr=subprocess.STDOUT,shell=True)
         print result
 
         #Check if the Output file exists
-        if (os.path.exists(output_file_path)):
-            print("Output file: ",output_file_path,"exists. SUCCESS!")
+        if (os.path.exists(output_xml_file_path)):
+            print("Output file: ",output_xml_file_path, "exists. SUCCESS!")
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:84]: Add space after commas in parameter lists. Do this throughout the file.

TIP: You can pass your Python code through `autopep8`. I suggest the following: `autopep8 --max-line-length 120 <file>`

File: sonic-utilities-tests/sku_create_test.py [hunk 5]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+
+        #Check if the Output file and the model file have same contents
+        if self.are_file_contents_same(output_config_db_file_path, model_config_db_file_path) == True:
+            print("Output file: ",output_config_db_file_path, " and model file: ",model_config_db_file_path, "contents are same. SUCCESS!")
+        else:
+            pytest.fail("Output file: {} and model file: {} contents are not same. FAILURE!".format(output_config_db_file_path, model_config_db_file_path))
+
+    def test_sku_port_split(self):
+        if (not os.path.exists(config_db_file)):
+            pytest.fail("Input config_db.json file does not exist. Exitting...")
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:140]: s/Exitting/Exiting

File: sonic-utilities-tests/sku_create_test.py [hunk 6]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+            return
+        else:
+             shutil.copyfile(config_db_file, port_split_config_db_output_file_path)
+        
+        if (not os.path.exists(model_config_db_file_path)):
+            pytest.fail("Input port_config.ini file does not exist. Exitting...")
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:146]: s/Exitting/Exiting

File: sonic-utilities-tests/sku_create_test.py [hunk 7]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+        else:
+            if eth18_instance != eth18_dict:
+                pytest.fail("Failure: Port split information not found in config_db.json file")
+                return
+
+        print("Success: Port split information found in config_db.json file")
+
+    def test_sku_port_unsplit(self):
+        if (not os.path.exists(config_db_file)):
+            pytest.fail("Input config_db.json file does not exist. Exitting...")
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:211]: s/Exitting/Exiting

File: sonic-utilities-tests/sku_create_test.py [hunk 8]
```diff
@@ -51,30 +69,197 @@ def are_file_contents_same(self,fname1,fname2):
...
+            return
+        else:
+             shutil.copyfile(config_db_file, port_unsplit_config_db_output_file_path)
+        
+        if (not os.path.exists(model_config_db_file_path)):
+            pytest.fail("Input port_config.ini file does not exist. Exitting...")
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_test.py:217]: s/Exitting/Exiting

**Review by jleveque — CHANGES_REQUESTED** [review]

File: sonic-utilities-tests/sku_create_test.py (hunk 1 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 2 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 3 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 4 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 5 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 6 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 7 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

File: sonic-utilities-tests/sku_create_test.py (hunk 8 above)
Comment (madhanmellanox) [reply sonic-utilities-tests/sku_create_test.py]: addressed in next commit

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

File: setup.py [hunk 9]
```diff
@@ -66,8 +66,15 @@
...
-                                  'sku_create_input/Mellanox-SN2700-D48C8/*']
+                                  'sku_create_input/2700_files/*',
+                                  'sku_create_input/3800_files/*',
+                                  'sku_create_input/port_split_files/*',
+                                  'sku_create_input/port_unsplit_files/*',
+                                  'sku_create_input/2700_files/ACS-MSN2700/*',
+                                  'sku_create_input/2700_files/Mellanox-SN2700-D48C8/*',
+                                  'sku_create_input/2700_files/Mellanox-SN2700-C28D8-ORIG/*',
+                                  'sku_create_input/3800_files/ACS-MSN3800/*',
+                                  'sku_create_input/3800_files/Mellanox-SN3800-D28C50/*']
```
Comment (jleveque) [inline setup.py:77]: This is OK for now, but hard to maintain. For the future we should look into a way of adding directories recursively, which is apparently not easy in setup.py. :(

Some discussion here for posterity: https://stackoverflow.com/questions/40031422/python-recursively-include-package-data-in-setup-py

File: sonic-utilities-tests/joe_test_folder/test [hunk 10]
```diff
@@ -0,0 +1 @@
+Creating this test file for git purposes
```
Comment (jleveque) [inline sonic-utilities-tests/joe_test_folder/test:1]: Delete this file

File: sonic-utilities-tests/sku_create_input/port_split_files/test [hunk 11]
```diff
@@ -0,0 +1 @@
+this is a test file for maintaining the port_split_files directory in git
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_input/port_split_files/test:1]: Delete this file?

File: sonic-utilities-tests/sku_create_input/port_unsplit_files/test [hunk 12]
```diff
@@ -0,0 +1 @@
+this is a test file for maintaining port_unsplit_files directory in git
```
Comment (jleveque) [inline sonic-utilities-tests/sku_create_input/port_unsplit_files/test:1]: Delete this file?

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by jleveque** [comment]
Tests failing on deleted files.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**